### PR TITLE
Add comments to `chain/network/src/test_utils.rs`

### LINKED ISSUE #5156: Write reference level documentation for the network code
To make network code easier to understand we should write a reference level documentation. 
That will make it easier for new developers to understand different parts of the networking code.

- [x] https://github.com/near/nearcore/pull/5580
- [x] https://github.com/near/nearcore/pull/5621
- [x] https://github.com/near/nearcore/pull/5622
- [x] https://github.com/near/nearcore/pull/5623
- [x] https://github.com/near/nearcore/pull/5625
- [x] https://github.com/near/nearcore/pull/5626
- [x] https://github.com/near/nearcore/pull/5627

Add in-code documentation
- [x] - `Graph` - https://github.com/near/nearcore/pull/5162
- [x] Add comments to `utils.rs` (https://github.com/near/nearcore/pull/5216)
- [x] Add comments to `test_utils.rs` (https://github.com/near/nearcore/pull/5217)
- [x] Add comments/refactor `rate_counter.rs` (https://github.com/near/nearcore/pull/5218)

Dead code
- [x] remove dead code (https://github.com/near/nearcore/pull/5207)

Improve code quality:
- [x]  Remove not needed clones - https://github.com/near/nearcore/pull/5209
- [x]  Remove 'pub' modifier from `Edge` - https://github.com/near/nearcore/pull/5223
- [x] Use constants for time in db.rs https://github.com/near/nearcore/pull/5215
- [x] Refactor Arc<Edge> to a new struct https://github.com/near/nearcore/pull/5228/files

### PR DESCRIPTION
**Pull request by pmnoxx**
Add comments to `chain/network/src/test_utils.rs`

Also rename `WaitOrTimeout` to `WaitOrTimeoutActor`.

Resolves #5156

### CODE REVIEW

File: chain/network/src/test_utils.rs [hunk 1]
```diff
@@ -310,7 +333,14 @@ pub fn make_peer_manager_routing_table_addr_pair(
     (peer_manager_addr, routing_table_addr)
 }
 
-#[allow(dead_code)]
+// Make peer manager for unit tests
+//
+// Returns:
+//    PeerManagerActor
+//    PeerId - PeerId associated with given actor
+//    Arc<AtomicUsize> - shared pointer for couting the number of receiver
```
Comment (mm-near) [inline chain/network/src/test_utils.rs:341]: s/receiver/received

File: chain/network/src/test_utils.rs [hunk 2]
```diff
@@ -237,6 +258,8 @@ impl Handler<StopSignal> for PeerManagerActor {
     }
 }
 
+/// Ban peer for unit tests.
+/// Calls `try_ban_peer` in `PeerManagerActor`.
```
Comment (mm-near) [inline chain/network/src/test_utils.rs:262]: shouldn't we have #[cfg(feature = "test_features")] on basically whole file?

**Review by mm-near — APPROVED** [review]

File: chain/network/src/test_utils.rs (hunk 1 above)
Comment (pmnoxx) [reply chain/network/src/test_utils.rs:341]: Fixed

File: chain/network/src/test_utils.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/network/src/test_utils.rs:262]: Some of the tests run without `test_features`  feature. So we can't do that unless we move all tests under `test_features` flag.

**Comment by pmnoxx** [comment]
@chefsale
@frol
@khorolets
Can you take a look?

**Review by chefsale — APPROVED** [review]

**Pull request merged**